<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from silia.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="silia" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" xmlns:xacro="http://ros.org/wiki/xacro">
  <!-- common stuff -->
  <material name="Blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="Green">
    <color rgba="0.0 0.8 0.0 1.0"/>
  </material>
  <material name="Grey">
    <color rgba="0.7 0.7 0.7 1.0"/>
  </material>
  <material name="Grey2">
    <color rgba="0.9 0.9 0.9 1.0"/>
  </material>
  <material name="Red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="Orange">
    <color rgba="1.0 0.5 0.0 1.0"/>
  </material>
  <material name="White">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <material name="Black">
    <color rgba="0.05 0.05 0.05 1.0"/>
  </material>
  <material name="Caster">
    <texture filename="package://pr2_ogre/Media/materials/textures/pr2_caster_texture.png"/>
  </material>
  <material name="Wheel_l">
    <texture filename="package://pr2_ogre/Media/materials/textures/pr2_wheel_left.png"/>
  </material>
  <material name="Wheel_r">
    <texture filename="package://pr2_ogre/Media/materials/textures/pr2_wheel_right.png"/>
  </material>
  <material name="RollLinks">
    <texture filename="package://pr2_ogre/Media/materials/textures/pr2_wheel_left.png"/>
  </material>
  <!-- arm mount positions | relative to base_link -->
  <!-- arm reference positions | used as calibration_rising for arm chain-->
  <!-- arm -->
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1.0"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.07"/>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
    </collision>
  </link>
  <!-- workcell -->
  <joint name="stack_joint" type="fixed">
    <origin rpy="0 0 0.785398163397" xyz="1 0.5 0"/>
    <parent link="base_link"/>
    <child link="stack_link"/>
  </joint>
  <link name="stack_link"/>
  <!-- composition of the robot -->
  <joint name="arm_0_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="arm_0_link"/>
  </joint>
  <link name="arm_0_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1.0"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!--cylinder radius="0.060" length="0.30"/-->
        <mesh filename="package://cob_kuka_description/meshes/kr16/joint_0.stl"/>
      </geometry>
      <material name="Black"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
    </collision>
  </link>
  <joint name="arm_1_joint" type="revolute">
    <origin rpy="3.14159265359 0 0" xyz="0 0 0.390"/>
    <parent link="arm_0_link"/>
    <child link="arm_1_link"/>
    <axis xyz="0 0 1"/>
    <calibration rising="0.0"/>
    <dynamics damping="10"/>
    <limit effort="200" lower="-2.9670" upper="2.9670" velocity="1.9634"/>
    <safety_controller k_position="20" k_velocity="50" soft_lower_limit="-2.957" soft_upper_limit="2.957"/>
  </joint>
  <link name="arm_1_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <!--mass value="5.9907"/-->
      <mass value="2.0"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cob_kuka_description/meshes/kr16/joint_1.stl"/>
        <!--cylinder radius="0.060" length="0.30"/-->
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!--mesh filename="package://cob_kuka_description/meshes/kr16/joint_1_collision.stl" /-->
        <cylinder length="0.30" radius="0.060"/>
      </geometry>
    </collision>
  </link>
  <joint name="arm_2_joint" type="revolute">
    <origin rpy="1.57079632679 0 0" xyz="0.260 0.07 -0.285"/>
    <parent link="arm_1_link"/>
    <child link="arm_2_link"/>
    <axis xyz="0 0 1"/>
    <calibration rising="0.0"/>
    <dynamics damping="10"/>
    <limit effort="200" lower="-2.0943" upper="2.0943" velocity="1.9634"/>
    <safety_controller k_position="20" k_velocity="50" soft_lower_limit="-2.0843" soft_upper_limit="2.0843"/>
  </joint>
  <link name="arm_2_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <!--mass value="5.734939"/-->
      <mass value="2.0"/>
      <inertia ixx="0.03" ixy="0" ixz="0" iyy="0.03" iyz="0" izz="0.03"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cob_kuka_description/meshes/kr16/joint_2.stl"/>
        <cylinder length="0.30" radius="0.060"/>
      </geometry>
      <material name="Black"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!--mesh filename="package://cob_kuka_description/meshes/kr16/joint_2_collision.stl"/-->
        <cylinder length="0.30" radius="0.060"/>
      </geometry>
    </collision>
  </link>
  <joint name="arm_3_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.680 0 -0.04"/>
    <parent link="arm_2_link"/>
    <child link="arm_3_link"/>
    <axis xyz="0 0 1"/>
    <calibration rising="0.0"/>
    <dynamics damping="5"/>
    <limit effort="100" lower="-2.9670" upper="2.9670" velocity="1.9634"/>
    <safety_controller k_position="20" k_velocity="25" soft_lower_limit="-2.957" soft_upper_limit="2.957"/>
  </joint>
  <link name="arm_3_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <!--mass value="4.214668"/-->
      <mass value="2.0"/>
      <inertia ixx="0.03" ixy="0" ixz="0" iyy="0.03" iyz="0" izz="0.03"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cob_kuka_description/meshes/kr16/joint_3.stl"/>
        <!--cylinder radius="0.060" length="0.30"/-->
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!--mesh filename="package://cob_kuka_description/meshes/kr16/joint_3_collision.stl"/-->
        <cylinder length="0.30" radius="0.060"/>
      </geometry>
    </collision>
  </link>
  <joint name="arm_4_joint" type="revolute">
    <origin rpy="0 -1.57079632679 0" xyz="0.400 0.035 0.118"/>
    <parent link="arm_3_link"/>
    <child link="arm_4_link"/>
    <axis xyz="0 0 1"/>
    <calibration rising="0.0"/>
    <dynamics damping="5"/>
    <limit effort="100" lower="-2.0943" upper="2.0943" velocity="1.9634"/>
    <!--limit effort="176" velocity="0.4189" upper="2.0943951" lower="-2.0943951"/-->
    <safety_controller k_position="20" k_velocity="25" soft_lower_limit="-2.0843" soft_upper_limit="2.0843"/>
  </joint>
  <link name="arm_4_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <!--mass value="3.669819"/-->
      <mass value="2.0"/>
      <inertia ixx="0.03" ixy="0" ixz="0" iyy="0.03" iyz="0" izz="0.03"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cob_kuka_description/meshes/kr16/joint_4.stl"/>
        <!--cylinder radius="0.060" length="0.30"/-->
      </geometry>
      <material name="Black"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!--mesh filename="package://cob_kuka_description/meshes/kr16/joint_4_collision.stl"/-->
        <cylinder length="0.30" radius="0.060"/>
      </geometry>
    </collision>
  </link>
  <joint name="arm_5_joint" type="revolute">
    <origin rpy="0 1.57079632679 0" xyz="0.1 0 -0.270"/>
    <parent link="arm_4_link"/>
    <child link="arm_5_link"/>
    <axis xyz="0 0 1"/>
    <calibration rising="0.0"/>
    <dynamics damping="5"/>
    <limit effort="100" lower="-2.9670" upper="2.9670" velocity="1.9634"/>
    <safety_controller k_position="20" k_velocity="25" soft_lower_limit="-2.957" soft_upper_limit="2.957"/>
  </joint>
  <link name="arm_5_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="2.313272"/>
      <inertia ixx="0.03" ixy="0" ixz="0" iyy="0.03" iyz="0" izz="0.03"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cob_kuka_description/meshes/kr16/joint_5.stl"/>
        <!--cylinder radius="0.060" length="0.30"/-->
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!--mesh filename="package://cob_kuka_description/meshes/kr16/joint_5_collision.stl"/-->
        <cylinder length="0.30" radius="0.060"/>
      </geometry>
    </collision>
  </link>
  <joint name="arm_6_joint" type="revolute">
    <origin rpy="0 -1.57079632679 0" xyz="0.158 0 -0.1"/>
    <!--origin xyz="0.115 0 0" rpy="0 -${M_PI/2} 0" /-->
    <parent link="arm_5_link"/>
    <child link="arm_6_link"/>
    <axis xyz="0 0 1"/>
    <calibration rising="0.0"/>
    <dynamics damping="5"/>
    <limit effort="30" lower="-2.0943951" upper="2.0943951" velocity="1.9634"/>
    <!--limit effort="20.1" velocity="0.4363" upper="2.0943951" lower="-2.0943951" /-->
    <safety_controller k_position="20" k_velocity="25" soft_lower_limit="-2.0843951" soft_upper_limit="2.0843951"/>
  </joint>
  <link name="arm_6_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="2.057856"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://cob_kuka_description/meshes/kr16/joint_6.stl"/>
        <!--cylinder radius="0.060" length="0.30"/-->
      </geometry>
      <material name="Black"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!--mesh filename="package://cob_kuka_description/meshes/kr16/joint_6_collision.stl"/-->
        <cylinder length="0.30" radius="0.060"/>
      </geometry>
    </collision>
  </link>
  <xacro:cob_kr16_gazebo name="arm"/>
  <xacro:cob_kr16_transmission name="arm"/>
</robot>

